x-pack/legacy/plugins/file_upload/public/util/pattern_reader.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const oboe = require('oboe');
+
+
+export class PatternReader {
+
+  constructor({ onFeatureDetect, onStreamComplete }) {
+    this._oboeStream = oboe();
+    this._onGeoJSONFeaturePatternDetect(onFeatureDetect);
```

Comment:
_onGeoJSONFeaturePatternDetect is not responding to feature patterns, instead its setting up the handlers. Maybe rename _onGeoJSONFeaturePatternDetect to something that better resembles what its doing, like registerFeaturePatternHandler.